### PR TITLE
[doc] Tiny followups to new getpeerinfo connection type field 

### PR DESCRIPTION
**Pull request by amitiuttarwar**
two commits addressing small followups from #19725 

* first commit adds a clarification in the release notes that this field shouldn't be expected to be stable (suggested by sdaftuar in https://github.com/bitcoin/bitcoin/pull/19725#issuecomment-697421878) 

* second commit moves the `CONNECTION_TYPE_DOC` object out of the header file to reduce the size of the binary (suggested by MarcoFalke in https://github.com/bitcoin/bitcoin/pull/19725#discussion_r495467895, he tested and found a decrease of 10kB)

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -108,6 +108,8 @@ will trigger BIP 125 (replace-by-fee) opt-in. (#11413)
 - The `getpeerinfo` RPC now returns a `connection_type` field. This indicates
   the type of connection established with the peer. It will return one of six
   options. For more information, see the `getpeerinfo` help documentation.
+  Please note that this string output is unlikely to be stable in upcoming
+  releases as we iterate to best capture connection behaviors.
```
Comment (MarcoFalke) [inline doc/release-notes.md:112]: This is a generally correct statement, that could go into the RPC help instead?

File: src/net.h [hunk 2]
```diff
@@ -116,13 +116,7 @@ struct CSerializedNetMsg
...
 };
 
-const std::vector<std::string> CONNECTION_TYPE_DOC{
-    "outbound-full-relay (default automatic connections)",
-    "block-relay-only (does not relay transactions or addresses)",
-    "inbound (initiated by the peer)",
-    "manual (added via addnode RPC or -addnode/-connect configuration options)",
-    "addr-fetch (short-lived automatic connection for soliciting addresses)",
-    "feeler (short-lived automatic connection for testing addresses)"};
+extern const std::vector<std::string> CONNECTION_TYPE_DOC;
```
Comment (promag) [inline src/net.h:119]: f476fbdf84c88b7ab90aa3e9240b39806525a8b6

not sure if this was asked before, `CONNECTION_TYPE_DOC` is used only in src/rpc/net.cpp, why not just have it there?

Comment (amitiuttarwar) [reply src/net.h]: hmm, good point! 

history: Marco suggested extracting in https://github.com/bitcoin/bitcoin/pull/19725#discussion_r470944955. I thought a benefit was to keep `CONNECTION_TYPE_DOC` next to `ConnectionType` definition for future updates, but now we don't have that regardless. I've moved into `src/rpc/net.cpp` and added a comment on the `ConnectionType` enum for future code changes.

File: doc/release-notes.md (hunk 1 above)
Comment (amitiuttarwar) [reply doc/release-notes.md]: done

**Comment by achow101** [comment]
ACK 41dca087b73a3627107603694f5a982ea2a53189

**Comment by laanwj** [comment]
Code review ACK 41dca087b73a3627107603694f5a982ea2a53189

**Pull request merged**